ui/src/alerting/reducers/checks.ts

```diff
@@ -49,16 +51,34 @@ export default (
 
       case 'SET_CURRENT_CHECK':
         draftState.current.status = action.payload.status
-        if (action.payload.check) {
-          draftState.current.check = action.payload.check
-        }
+        draftState.current.check = action.payload.check
```

Comment:
How useful is immer in this reducer? It seems like immer really shines for deeply nested state objects. But it also risks introducing hard to understand bugs due to it's [limitations](https://github.com/immerjs/immer#pitfalls). I wonder if we should only be using it when its benefits outway its costs.